File: leetcode_121/solution.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let n = prices.length;
    if (n == 0) {
        return 0;
    } else {
        let _min = prices[0], profit = 0;
        for (let i = 1; i < n; i++) {
            if (prices[i] < _min) {
                _min = prices[i];
            } else {
                let delta = prices[i] - _min;
                if (delta > profit) {
                    profit = delta;
                }
            }
        }
        return profit;
    }
};

File: leetcode_121/solution.go
package main;

func maxProfit(prices []int) int {
  n := len(prices)
  if n == 0 {
    return 0
  } else {
    _min, profit := prices[0], 0
    for i := 1; i < n; i += 1 {
      if prices[i] < _min {
        _min = prices[i]
      } else {
        delta := prices[i] - _min
        if delta > profit {
          profit = delta
        }
      }
    }
    return profit
  }
}
